api_client.py

```diff
@@ -38,3 +44,116 @@ def get_authorization(securex_host_name, secuerx_client_id, securex_client_passw
+                                        securex_host_name,
+                                        secuerx_client_id,
+                                        securex_client_password
+                                    ),
+                                    "Accept": "application/json"
+                                })
+        me_response = response.json()
+        customer_id = me_response["identity"]["customerId"]
+
+    return customer_id
+
+
+class CollectionIterator:
+    """
+    Object designed to help with processing of paged collections based on iterator pattern.
+    Both GET and POST collections are supported.
+    Params:
+    - collection_url_path (str) - URL path of collection we'd like to get items (without hostname and query parameters),
+      e.g., "/alert-management/customer/customerId/alerts"
+    - authorization_fn (str) - function able to generate "Authorization" header,
+      use "get_authorization" wrapped with credentials
+    - (optional) query_params (key: str - value: str) - query parameters in dictionary
+      e.g., { "size": "100" }
+    - (optional) cursor (str) - position of item (cursor) in collection where to continue iterating (usually cursor of last item from previous run)
+      e.g., "123456"
+    - (optional) request_body (dict) - do POST with this provided requestBody
+    """
+    def __init__(self, collection_url_path, authorization_fn, query_params=None, cursor=None, request_body=None):
+        # keep immutable these constructor parameters
+        self.collection_url_path = collection_url_path
+        self.authorization_fn = authorization_fn
+        self.query_params = query_params or {}
+        self.request_body = request_body
+
+        self.has_api_response = False
+        self.has_next_page = None
+        self.end_cursor = cursor
+        self.next_page_url = self.__build_next_page_url_from_variables(cursor)
+        self.current_page_items = []
+        self.current_page_items_pointer = 0
+
+    def has_next(self):
+        if not self.has_api_response:
+            self.__fetch_next_page()
+
+        return self.__has_current_page_items() or self.has_next_page
+
+    def next(self):
+        if not self.has_api_response \
+                or (not self.__has_current_page_items() and self.has_next_page):
+            self.__fetch_next_page()
+
+        if self.__has_current_page_items():
+            item = self.current_page_items[self.current_page_items_pointer]
+            self.current_page_items_pointer = self.current_page_items_pointer + 1
+            return item
+        else:
+            return None
+
+    def __fetch_next_page(self):
+        headers = {
+            "Authorization": self.authorization_fn(),
+            "Accept": "application/json"
+        }
+
+        if self.request_body:
+            response = requests.post(API_HOST_NAME + self.next_page_url, headers=headers, json=self.request_body)
```

Comment:
consider using the session object: https://docs.python-requests.org/en/master/user/advanced/#session-objects

`requests.post|get` creates a new session on each method call.